lib/zhr_devs/identity_management/identity.ex

```diff
@@ -26,6 +26,18 @@ defmodule ZhrDevs.Identity do
     GenServer.start_link(__MODULE__, success_struct, name: via_tuple(hashed_identity))
   end
 
+  def renew_login(hashed_identity) when is_binary(hashed_identity) do
+    GenServer.call(via_tuple(hashed_identity), :renew_login)
+  end
+
+  @spec parse_hashed_identity(String.t()) :: Uptight.Base.Urlsafe.t()
+  def parse_hashed_identity(hashed_identity) do
+    hashed_identity = Uptight.Base.mk_url(hashed_identity)
+    assert Result.is_ok?(hashed_identity), "Hashed identity is not valid"
```

Comment:
More details here. I think the only way this fails is if the input isn't binary().